app/models/event.rb

```diff
@@ -1,12 +1,44 @@
 class Event < ApplicationRecord
+  include ApplicationHelper
+
   validates :title, presence: true, length: { maximum: 50 }
   validates :description, presence: true, length: { maximum: 200 }
   validates :start_at, presence: true
   validates :venue, presence: true, length: { maximum: 50 }
   validates :source_link, presence: true
 
+  after_create_commit :notify
+
+  class << self
+    def notify_upcoming_events
+      events = Event.where(start_at: Date.today..Date.today.days_since(2).end_of_day)
+      Notification.new.notify_events(events:, content: "近日開催のイベントをお知らせ")
+    end
+
+    def notify_events_in_this_week
+      events = Event.where(start_at: Date.today.all_week)
```

Comment:
```suggestion
      events = Event.where(start_at: Time.current.all_week)
```

自分で書いたコードに対してアレだけど、こう書かなきゃダメかも…？

Reply:
たしかに、 `start_at` 自体が日付だけでなく日"時"を表すdatetime型なので、 Dateより Time型を使う方が合っている気がしますが、そういう意図でコメントされたってことで合ってますかね??

Reply:
発行されるクエリのちがいを見て、今回の用途に合うものを採用してもらいたいです :pray: :dash:

Reply:
分かりました! クエリ確認しますね。明日以降やってみます!

Reply:
クエリの違いを確認しました。

- Dateを使った場合
```sql
Event.where(start_at: Date.today.all_week)
  Event Load (11.9ms)  SELECT "events".* FROM "events" WHERE "events"."start_at" BETWEEN '2024-12-02' AND '2024-12-08' /* loading for pp */ LIMIT 11 /*application='Manabiya'*/
```

- Timeを使った場合
```sql
> Event.where(start_at: Time.current.all_week)
  Event Load (5.5ms)  SELECT "events".* FROM "events" WHERE "events"."start_at" BETWEEN '2024-12-01 15:00:00' AND '2024-12-08 14:59:59.999999' /* loading for pp */ LIMIT 11 /*application='Manabiya'*/ # DBにはUTC時刻で保存されてるので15:00~になっている

```

`start_at` はdatetime型で、日付だけでなく時刻ももつデータ型なので、条件検索する時に時刻の情報も持っているTimeクラスを使う方が自然・適切と思ったので、直していきます!

Reply:
コミットつみましたので、ご確認をお願いします!
- https://github.com/teacherteacher-jp/manabiya/pull/108/commits/b162dff896864a5045a7d01327f2fe672a92e2ab